File: algorithms/search/binarySearch.test.js
import binarySearch from './binarySearch.js'

const testBinarySearch = () => {
    const nums = [5, -2, 12, 0, 9]

    let idx = binarySearch(nums, 12), expected = 2
    console.assert(idx === expected, `expect to be: ${expected}, got instead: ${idx}`)

    idx = binarySearch(nums, 6), expected = -1
    console.assert(idx === expected, `expect to be: ${expected}, got instead: ${idx}`)
}

testBinarySearch()

File: algorithms/search/binary_search_test.go
package search

import (
	"testing"
)

func TestBinarySearchInt(t *testing.T) {
	var nums = []int{-2, 0, 5, 9, 12}

	target := 9
	if idx, expected := BinarySearchInt(nums, target), 3; idx != expected {
		t.Errorf("expect binarySearch(..., %d) to be: %d, got instead: %d", target, expected, idx)
	}

	target = -2
	if idx, expected := BinarySearchInt(nums, target), 0; idx != expected {
		t.Errorf("expect binarySearch(..., %d) to be: %d, got instead: %d", target, expected, idx)
	}

	target = 6
	if idx, expected := BinarySearchInt(nums, target), -1; idx != expected {
		t.Errorf("expect binarySearch(..., %d) to be: %d, got instead: %d", target, expected, idx)
	}
}
